src/registrar/models/domain_application.py

```diff
@@ -147,12 +147,12 @@ class DomainApplication(TimeStampedModel):
         related_name="current+",
     )
 
-    requested_domain = models.ForeignKey(
-        Website,
+    requested_domain = models.OneToOneField(
```

Comment:
**question (non-blocking)**: I wonder if a `OneToOneField` might not be right here. @h-m-f-t has mentioned before that we should be ready for multiple applications for the same domain name to be opened, then one of them could be awarded. I think that this could still work here since the `name` field of `Domain` is not constrained to be `unique`, but I'm also not sure if that is how we want it or not.

**suggestion (non-blocking)**: Could we maybe have a test case of the models to explicitly show what we want to be possible? Create two domain applications for the same domain at the same time? I know that we do it in one of the tests, but since that test is about something else, it isn't as clear to me that we are defining in the tests how we specifically want it to be possible.

Reply:
1. I see what you are saying. I think `OneToOneField` is the right choice for now because we've agreed that an applicant can only apply for one domain at a time (one domain per application) and, while a domain _string_ may not be unique during the application phase, the Domain model and its associated Foreign Key models track additional information, like nameserver data, which is uniquely tied to one organization (one application per domain).

1. Done-ish. I've sketched out some test cases for the interaction between the DomainApplication model and the Domain model, but have made use of `@skip` on them, since there's no working implementation for it.